### DISCUSSION TITLE
Is anyone having troubles trying to add a dependency that was just released?

### DISCUSSION THREAD

**Discussion by fedecalendino** [python-poetry/poetry#6064] [Q&A] [answered]
Hi!

I've been struggling with this since yesterday, maybe you can help me out.

Last night I released one of my libraries to pypi, but no matter what I do poetry doesn't seem to find the latest version. I tried again bumping the version just in case, but nothing. The odd thing is that I have no issues installing them with pip.

**alfred-pyflow** [github](https://pypi.org/project/alfred-pyflow/) / [pypi](https://pypi.org/project/alfred-pyflow/)
```bash
$ poetry add alfred-pyflow@1.0.0
Could not find a matching version of package alfred-pyflow

---

$ pip install alfred-pyflow==1.0.0
Collecting alfred-pyflow==1.0.0
    ...
```

I started to try it the latest releases on pypi to see if happened with other releases, and some do and some don't.

**skit-labels** [github](https://github.com/skit-ai/skit-labels) / [pypi](https://pypi.org/project/skit-labels/)
```bash
$ poetry add skit-labels==0.3.25
Could not find a matching version of package skit-labels

---

$ pip install skit-labels==0.3.25
Collecting skit-labels==0.3.25
    ...
```

**re-data** [github](https://github.com/re-data/re-data) [pypi](https://pypi.org/project/re-data/)
```bash
$ poetry add re-data@0.9.2
    ...
    ...
  • Installing re-data (0.9.2)

---

$ pip install re-data==0.9.2
Collecting re-data==0.9.2
    ...
```

**expressmoney-points** [pypi](https://pypi.org/project/expressmoney-points/)
```bash
poetry add expressmoney-points@1.18.1
Could not find a matching version of package expressmoney-points

---

$ pip install expressmoney-points==1.18.1
Collecting expressmoney-points==1.18.1
    ...
```

**checkov** [github](https://github.com/bridgecrewio/checkov) / [pypi](https://pypi.org/project/checkov/)
```bash
poetry add checkov@2.1.72
Could not find a matching version of package checkov

---

@ pip install checkov==2.1.72
Collecting checkov==2.1.72
    ...
```

---

My current version of poetry is **1.2.0b3**, tho I had **1.1.14** before but I installed the preview version hoping it would fix the issue but no luck.

Also, clearing poetry's pypi cache didn't help either.

**Comment by neersighted** [comment, marked as answer]
See #3543 and https://github.com/pypi/warehouse/issues/11939 -- this is an upstream PyPI issue that does not affect pip because of the APIs it uses. There is no resolution for now, but please weigh in on the issue if you think Poetry can communicate what is happening better.
